src/sid/statistics.py

```diff
@@ -76,12 +82,14 @@ def calculate_r_zero(df: pd.DataFrame, window_length: int = 7) -> pd.Series:
     """
     r_effective = calculate_r_effective(df=df, window_length=window_length)
 
+    df["immune"] = df["immunity"] > threshold
+
     grouper = _create_time_grouper(df)
     if grouper is None:
-        share_susceptibles = 1 - df["immunity"].mean()
+        share_susceptibles = 1 - df["immune"].mean()
         r_zero = r_effective / share_susceptibles
     else:
-        share_susceptibles = 1 - df.groupby(grouper)["immunity"].mean()
+        share_susceptibles = 1 - df.groupby(grouper)["immune"].mean()
```

Comment:
I think `not_susceptible.groupby(df[grouper])` (or sth similar) should work here.